runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -87,6 +88,25 @@ public synchronized void reset() {
      * {@inheritDoc}
      */
     @Override
+    public synchronized void seek(long globalAddress) {
+        // pop any stream context which has a max address
+        // less than the global address
+        while (this.streamContexts.size() > 1) {
+            if (this.streamContexts.first().maxGlobalAddress <
+                    globalAddress)
+            {
+                this.streamContexts.pollFirst();
+            }
+        }
+
+        // now request a seek on the context
+        this.streamContexts.first().seek(globalAddress);
```

Comment:
So we are certain it will be single context here, as all other will be removed,  and only one will have maxGlobalAddress=globalAddress, correct?